backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/DemoOktaRepository.java

```diff
@@ -120,12 +117,22 @@ public Optional<OrganizationRoleClaims> updateUser(IdentityAttributes userIdenti
   }
 
   public Optional<OrganizationRoleClaims> updateUserEmail(
-      IdentityAttributes userIdentity, String email) {
-    if (!usernameOrgRolesMap.containsKey(userIdentity.getUsername())) {
+      IdentityAttributes userIdentity, String newEmail) {
+    String currentEmail = userIdentity.getUsername();
+    if (!usernameOrgRolesMap.containsKey(currentEmail)) {
       throw new IllegalGraphqlArgumentException(
           "Cannot change email of Okta user with unrecognized username");
     }
-    OrganizationRoleClaims orgRoles = usernameOrgRolesMap.get(userIdentity.getUsername());
+
+    if (usernameOrgRolesMap.containsKey(newEmail)) {
+      throw new ConflictingUserException();
+    }
+
+    String org = usernameOrgRolesMap.get(userIdentity.getUsername()).getOrganizationExternalId();
```

Comment:
⚠️ Should I be adding the old username to `inactiveUsernames` or something like that instead of just removing it 🤔

Reply:
I don't think that's necessary! In real Okta if a username is updated, we don't hold on to the old one anywhere.

Reply:
Ok!! Thank you for confirming!! :D